pandas/core/arrays/sparse.py

```diff
@@ -2025,3 +2056,126 @@ def to_coo(self, row_levels=(0, ), column_levels=(1, ), sort_labels=False):
                                                  column_levels,
                                                  sort_labels=sort_labels)
         return A, rows, columns
+
+    def to_dense(self):
+        from pandas import Series
+        return Series(self._parent.array.to_dense(),
+                      index=self._parent.index,
+                      name=self._parent.name)
+
+
+class SparseFrameAccessor(PandasDelegate):
+
+    def __init__(self, data=None):
+        # Store the Series since we need that for to_coo
+        self._parent = data
+        self._validate(data)
+
+    def _validate(self, data):
+        dtypes = data.dtypes
+        if not all(isinstance(t, SparseDtype) for t in dtypes):
+            raise AttributeError(_validation_msg)
+
+    @classmethod
+    def from_spmatrix(cls, data, index=None, columns=None):
```

Comment:
Compare with `SparseDataFrame._init_spmatrix` at https://github.com/pandas-dev/pandas/blob/21769e9622630ab12bdcb480da0779eea644b04a/pandas/core/sparse/frame.py#L205

I'm need to do some benchmarking, but generally I find this easier to follow.

Reply:
No material difference with the SparseDataFrame constructor.

```python
In [29]: mat = scipy.sparse.random(10000, 100)

In [30]: %timeit pd.SparseDataFrame(mat, default_fill_value=0.0)
48.5 ms ± 1.15 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [31]: %timeit pd.DataFrame.sparse.from_spmatrix(mat)
44.6 ms ± 3.54 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

```